### ISSUE TITLE
No bottle found for python on big sur

### ISSUE THREAD

**Issue by bavardage** [Homebrew/homebrew-core#67978]
# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ~~ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?~~
```% brew gist-logs python
Error: No logs.
```
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```
% brew doctor
Your system is ready to brew.
```

```
% brew config
HOMEBREW_VERSION: 2.7.1-54-g8945803
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8945803432764e6941ebbb55485fdf346d640ad0
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f2cc04c45e0fde4973784f5376e061aa0ccff24a
Core tap last commit: 47 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-x86_64
CLT: N/A
Xcode: 12.3
```

## What you were trying to do (and why)

Install python. The existing python install is the mac built-in one, I think.

## What happened (include command output)

<details>
  <summary>brew install python</summary>
  <pre>
Error: python@3.9: no bottle available!
You can try to install from source with e.g.
  brew install --build-from-source python@3.9
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
  </pre>
</details>

## What you expected to happen

I expect there to be a bottle available to use - at least https://formulae.brew.sh/formula/python@3.9 suggests there's a bottle available for big sur.

## Step-by-step reproduction instructions (by running `brew install` commands)

See above.

**Comment by carlocab** [comment]
You don't have the Command Line Tools installed:
```
CLT: N/A
```
To fix that, run
```
xcode-select --install
```
Note that this is exactly what `brew doctor` would've told you to do. Please make sure to actually do what is listed before ticking the checkbox next time. Thanks!

**Comment by carlocab** [comment]
Actually, now that I think about it, it's weird that you don't have the CLT installed but `brew doctor` isn't complaining. Reopening this for visibility.

**Comment by bavardage** [comment]
Thanks!

I can confirm that this works after installing the CLT.

**Comment by carlocab** [comment]
Yes. It seems that `brew doctor` doesn't complain about the absence of the CLT by design when you have Xcode installed. However, the error message from doing `brew install python@3.9` when you don't have the CLT installed could be more informative.

**Comment by rami3l** [comment]
I did encounter the same thing while trying to install `YouTube-dl` and `brew` tries to install `python` as a dependency. No CLT, `brew doctor` says nothing, ended up building CPython from source.

**Comment by carlocab** [comment]
Do you have Xcode installed? If so, then `brew doctor` saying nothing is by design, and I think perfectly fine. I think what could be improved here is the message you get from brew refusing to pour the `python@3.9` bottle.

**Comment by rami3l** [comment]
@carlocab Yes, I do have Xcode, but just being curious, is there anything in pouring the bottle that specifically requires CLT? I used to think that an Xcode install can cover all the CLT usage...

**Comment by carlocab** [comment]
Yes, there is:

https://github.com/Homebrew/homebrew-core/blob/644fcd4cc6e3ae975a3b945bf891d0a3015907e1/Formula/python%403.9.rb#L21-L32

`brew` is no longer displaying the message from the formula for some reason.

**Comment by rami3l** [comment]
@carlocab Thanks a lot, I can confirm that installing CLT solves this problem 🎉
